swap_meet/vendor.py

```diff
@@ -1,2 +1,105 @@
+
+from .item import Item
+
+
+
+# WAVE 1
 class Vendor:
-    pass
+    # constructor that is by default empty, but checks for input
+    def __init__(self, inventory = [] ):
+        if inventory :
+            self.inventory = inventory
+        else:
+            self.inventory = []
+
+    # method thats adds an item to inventory
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    # method that removes an item from inventory
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    # WAVE 2
+    # method that adds items that are the same category to a list
+    def get_by_category(self, category):
+        category_list = []
+        for item in self.inventory:
+            if item.category == category:
+                category_list.append(item)
+        return category_list
+
+    # WAVE 3
+    # swaps items from one inventory to another
+    def swap_items(self, another_vendor, my_item, their_item):
+        if  my_item not in self.inventory or their_item not in another_vendor.inventory:
+            return False
+        else:
+            self.inventory.remove(my_item)
+            another_vendor.inventory.append(my_item)
+            another_vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
+
+    # WAVE 4
+    def swap_first_item(self, another_vendor):
+        if  self.inventory == [] or another_vendor.inventory == []:
+            return False
+        else:
+            another_vendor.inventory.append(self.inventory[0])
+            self.inventory.remove(self.inventory[0])
+            self.inventory.append(another_vendor.inventory[0])
+            another_vendor.inventory.remove(another_vendor.inventory[0])
+            
+            return True
```

Comment:
See my comment above about `else` clauses. We could have reused the `swap_items` method to do the actual swapping.
```suggestion
    def swap_first_item(self, another_vendor):
        if  self.inventory == [] or another_vendor.inventory == []:
            return False
        
        self.swap_items(self, another_vendor, self.inventory[0], another_vendor.inventory[0])
            
            return True
```

This method would have a worst-case time complexity of O(n) due to `remove`.

Reply:
We can improve the time complexity to O(1) by using multiple assignment. 

```
        self.inventory[0], friend_vendor.inventory[0] = friend_vendor.inventory[0], self.inventory[0]
```        

Here's more info on Python multiple assignment: https://www.w3schools.com/python/gloss_python_assign_value_to_multiple_variables.asp